hooks/brc20/useBrc20TransferFees.ts

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+
+import { UTXO } from 'types';
+import { brc20TransferEstimateFees } from '../../transactions/brc20';
+
+type CommitValueBreakdown = {
+  commitChainFee: number;
+  revealChainFee: number;
+  revealServiceFee: number;
+  transferChainFee: number;
+  transferUtxoValue: number;
+};
+
+export enum ErrorCode {
+  INSUFFICIENT_FUNDS = 'INSUFFICIENT_FUNDS',
+  INVALID_TICK = 'INVALID_TICK',
+  INVALID_AMOUNT = 'INVALID_AMOUNT',
+  INVALID_FEE_RATE = 'INVALID_FEE_RATE',
+  SERVER_ERROR = 'SERVER_ERROR',
+}
+
+/**
+ * Estimates the fees for a BRC-20 1-step transfer
+ * @param addressUtxos - The UTXOs in the bitcoin address which will be used for payment
+ * @param tick - The 4 letter BRC-20 token name
+ * @param amount - The amount of the BRC-20 token to transfer
+ * @param feeRate - The desired fee rate for the transactions
+ * @param revealAddress - The address where the balance of the BRC-20 token lives. This is usually the ordinals address.
+ */
+const useBrc20TransferFees = (
+  addressUtxos: UTXO[],
+  tick: string,
+  amount: number,
+  feeRate: number,
+  revealAddress: string,
+) => {
+  const [commitValue, setCommitValue] = useState<number | undefined>();
+  const [commitValueBreakdown, setCommitValueBreakdown] = useState<CommitValueBreakdown | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorCode, setErrorCode] = useState<ErrorCode | undefined>();
+
+  useEffect(() => {
```

Comment:
consider exposing a method to execute this code instead running it on the hook call. and since you don't need to update a progress state, you can just return an error or the fees values. no need to use state.

Reply:
@victorkirov what was the conclusion on this?